prboom2/src/gl_main.c

```diff
@@ -2410,6 +2410,7 @@ void gld_ProjectSprite(mobj_t* thing, int lightlevel)
     scene_has_overlapped_sprites = true;
 
   sprite.index = gl_spriteindex++;
+  sprite.id = thing->index;
```

Comment:
The thing index is the incrementing value in the wad data, but it isn't a unique identifier. Every entity that spawns in the middle of a map has index `-1`. Maybe cast the thing address?

Reply:
This revealed some problems with handing large integers to the shader to convert to floats internally, so I had to rework things a bit, but I'm now using the address.